src/pages/settings/tabs/currencies/edit-currency.vue

```diff
@@ -123,10 +142,40 @@ export default defineComponent({
       },
     );
 
+    onMounted(() => {
+      isChecked.value = props.currency.custom
+    })
+
     const onDeleteHandler = () => {
       emit('delete');
     };
 
+    const deleteExchangeRates = async () => {
+      try {
+        await Promise.all([
+          deleteCustomRate([
+            {
+              baseCode: props.currency.code,
+              quoteCode: props.currency.quoteCode,
+            },
+            {
+              baseCode: props.currency.quoteCode,
+              quoteCode: props.currency.code,
+            },
+          ]),
+            currenciesStore.loadCurrencies(),
+        ]);
```

Comment:
I don't understand the need for `Promise.all` here. If `loadCurrencies` will fail, the interface will behave as it was custom rates deletion broken. Also, `Promise.all` makes calls in parallel, so it might be the case that rates were removed, but `loadCurrencies` will return old data just because it was actually called before rates removal. Just use await for each call one by one.

Also, `loadCurrencies`, loads both user-specific and generic (all world's existing) currencies. Generic currencies are never changed and should only be loaded once, on app initialization. Otherwise, useless 15KB of network bandwidth will be used.

<img width="457" alt="image" src="https://github.com/letehaha/budget-tracker-fe/assets/12257282/84c57370-2353-401f-959e-1d938fa37c57">